openedx/adg/lms/applications/views.py

```diff
@@ -291,99 +263,174 @@ def get_context_data(self, request):
             'user_application_id': user_application.id,
             'is_work_experience_not_applicable': user_application.is_work_experience_not_applicable,
             'is_education_experience_completed': user_application.is_education_experience_completed,
-        })
+            'background_form': form,
+        }
 
-        return context
+        return render(request, self.template_name, context)
+
+    def handle_redirection(self, request, form):
+        """
+        Redirects to contact info page on clicking back button and to business line page
+        on clicking next button
+
+        Returns:
+            HttpResponse object.
+        """
+        next_or_back_clicked = form.data.get('next_or_back_clicked')
+        if next_or_back_clicked:
+            if next_or_back_clicked == 'back':
+                return redirect('application_contact')
+            else:
+                return redirect('application_business_line_interest')
 
     def is_precondition_satisfied(self):
         """
-        Checks if a user's application is started but not completed.
+        Checks if a user's application is started but not completed. Also checks if the user
+        is a saudi national with an added birthdate
 
         Returns:
             Boolean, True or False.
         """
         application_hub = getattr(self.request.user, 'application_hub', None)
+        extended_profile = getattr(self.request.user, 'extended_profile', None)
 
-        return (
+        application_started_but_not_submitted = (
             application_hub and
             application_hub.is_written_application_started and
             not application_hub.is_written_application_completed
         )
 
+        has_valid_profile = extended_profile and extended_profile.is_saudi_national_and_has_birthdate
+
+        return application_started_but_not_submitted and has_valid_profile
+
     def handle_no_permission(self):
         """
-        Redirects to application hub.
+        Redirects to application hub for get request and returns a 400 error on post request
         """
-        return redirect('application_hub')
+        if self.request.method == 'POST':
+            return HttpResponse(status=HTTP_400_BAD_REQUEST)
+        else:
+            return redirect('application_hub')
 
 
-class CoverLetterView(RedirectToLoginOrRelevantPageMixin, View):
+class BusinessLineInterestView(RedirectToLoginOrRelevantPageMixin, View):
     """
-    View enabling the user to select a Business Line and upload or write a cover letter.
+    View enabling the user to select a Business Line and write their reason for interest in business.
     """
 
-    template_name = 'adg/lms/applications/cover_letter.html'
+    template_name = 'adg/lms/applications/business_line_interest.html'
     login_url = reverse_lazy('register_user')
 
+    def is_precondition_satisfied(self):
+        """
+        Checks if a user's education, work experience and background question have been added
+
```

Comment:
Can we please add a little more detail? when the pre condition is satisfied and when it is not?